core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -220,12 +221,64 @@ class ControllerApis(val requestChannel: RequestChannel,
   def handleBrokerRegistration(request: RequestChannel.Request): Unit = {
     val registrationRequest = request.body[BrokerRegistrationRequest]
     apisUtils.authorizeClusterOperation(request, CLUSTER_ACTION)
-    controller.registerBroker(registrationRequest.data())
+
+    controller.registerBroker(registrationRequest.data).handle[Unit]((reply, ex) => {
+      var err = Errors.NONE
```

Comment:
if statements can return a value in scala, so you don't need to mutate this

Reply:
Nice! Fixed.